screens/Nearby.js

```diff
@@ -0,0 +1,116 @@
+import React from "react";
+import { connect } from "react-redux";
+import { View, ScrollView, Text, TouchableOpacity } from "react-native";
+import { reusableStyles, specificStyles } from "../styles";
+import { setLandmark } from "../store/selectedLandmark";
+import { convertDistance, getDistance, orderByDistance } from "geolib";
+import { roundToOneDecimalPlace } from "../util/index"
+
+class Nearby extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentMapRegion: {},
+    };
+  }
+
+  componentDidMount () {
+    const currentMapRegion = this.props.mapRegion ?
+      this.props.mapRegion : {
+        latitude: 40.673868,
+        longitude: -73.970089,
+      };
+    this.setState({
+      currentMapRegion,
+    })
+  }
+
+  componentDidUpdate() {
+    if (this.state.currentMapRegion !== this.props.mapRegion) {
+      this.setState({
+        currentMapRegion: this.props.mapRegion,
+      });
+    }
+  }
+
+  reformatLandmarkData ( landmarks ) {
+    const { currentMapRegion } = this.state
+
+    return landmarks.map(landmark => {
+      let distance = getDistance({
+        latitude: currentMapRegion.latitude,
+        longitude: currentMapRegion.longitude
+      }, {
+        latitude: landmark.coordinate.latitude,
+        longitude: landmark.coordinate.longitude
+      });
+      distance = convertDistance(distance, 'mi')
+      return {
+        ...landmark,
+        latitude: landmark.coordinate.latitude,
+        longitude: landmark.coordinate.longitude,
+        distance: roundToOneDecimalPlace( distance ),
+      };
+    });
+  }
+
+  sortLandmarks (landmarks) {
+    let orderedLandmarks = (orderByDistance(this.state.currentMapRegion, landmarks)).slice(0, 7);
+
+    //prevents the user from seeng the landmark they've selected
+    //as the first nearby landmark
+    orderedLandmarks = orderedLandmarks[0].distance == 0 ? orderedLandmarks.slice(1, 7) : orderedLandmarks.slice(0, 6);
+
+    return orderedLandmarks;
+  }
+
+  render() {
+    const { landmarks } = this.props
+    const { currentMapRegion } = this.state
+
+    if (landmarks.length && currentMapRegion.latitude) {
```

Comment:
think about adding some `!!` in front of these so they're converted to actual booleans